crates/release_plz_core/src/command/update/package_dependencies.rs

```diff
@@ -20,17 +21,20 @@ impl PackageDependencies for Package {
         updated_packages: &'a [(&Package, Version)],
         workspace_dependencies: Option<&dyn TableLike>,
         workspace_dir: &Utf8Path,
-    ) -> anyhow::Result<Vec<&'a Package>> {
+    ) -> anyhow::Result<HashMap<&'a Package, SemverCheck>> {
         // Look into the toml manifest because `cargo_metadata` doesn't distinguish between
         // empty `version` in Cargo.toml and `version = "*"`
         let package_manifest = LocalManifest::try_new(&self.manifest_path)?;
         let package_dir = crate::manifest_dir(&package_manifest.path)?.to_owned();
 
-        let mut deps_to_update: Vec<&Package> = vec![];
-        for (p, next_ver) in updated_packages {
-            let canonical_path = p.canonical_path()?;
-            // Find the dependencies that have the same path as the updated package.
-            let matching_deps = package_manifest
+        updated_packages
+            .iter()
+            .filter_map(|(p, next_ver)| {
+                let Ok(canonical_path) = p.canonical_path() else {
```

Comment:
the code used `?` to report the error to the user. 
Instead here you are discarding the error and you don't even log it.
Please keep throwing the error. E.g. using `?` again. Same later in `should_update_dependency(dep, next_ver)?`